sklearn/utils/estimator_checks.py

```diff
@@ -312,6 +314,14 @@ def set_testing_parameters(estimator):
         if not isinstance(estimator, ProjectedGradientNMF):
             estimator.set_params(solver='cd')
 
+    if hasattr(estimator, "n_components"):
```

Comment:
Okay, I'll move that back to the concrete test. 

How did you check that, btw?